### ISSUE TITLE
Forward Mode Fabs call error

### ISSUE THREAD

**Issue by wsmoses** [wsmoses/Enzyme#394]
```llvm
define internal fastcc double @fwddiffejulia_mybesselj_17692(i64 signext %0, double %1, double %"'", double %2) unnamed_addr !dbg !215 {
top:
  store double %"'", double* %"'de", align 8
  %_replacementA = phi {}***
  %3 = fmul double %1, 5.000000e-01, !dbg !216
  %4 = load double, double* %"'de", align 8, !dbg !219
  %5 = fmul fast double %4, 5.000000e-01, !dbg !219
  store double %5, double* %"'de1", align 8, !dbg !219
  %6 = load double, double* %"'de1", align 8, !dbg !219
  switch i64 %0, label %L24 [
    i64 -1, label %L28
    i64 0, label %L44
    i64 1, label %L40
    i64 2, label %L14
    i64 3, label %L18
  ], !dbg !219

L14:                                              ; preds = %top
  %7 = fmul double %3, %3, !dbg !220
  %8 = load double, double* %"'de1", align 8, !dbg !221
  %9 = load double, double* %"'de1", align 8, !dbg !221
  %10 = fmul fast double %8, %3, !dbg !221
  %11 = fmul fast double %9, %3, !dbg !221
  %12 = fadd fast double %10, %11, !dbg !221
  store double %12, double* %"'de2", align 8, !dbg !221
  %13 = load double, double* %"'de2", align 8, !dbg !221
  br label %L40, !dbg !221

L18:                                              ; preds = %top
  %14 = fmul double %3, %3, !dbg !222
  %15 = load double, double* %"'de1", align 8, !dbg !222
  %16 = load double, double* %"'de1", align 8, !dbg !222
  %17 = fmul fast double %15, %3, !dbg !222
  %18 = fmul fast double %16, %3, !dbg !222
  %19 = fadd fast double %17, %18, !dbg !222
  store double %19, double* %"'de3", align 8, !dbg !222
  %20 = fmul double %3, %14, !dbg !222
  %21 = load double, double* %"'de1", align 8, !dbg !224
  %22 = load double, double* %"'de3", align 8, !dbg !224
  %23 = fmul fast double %21, %14, !dbg !224
  %24 = fmul fast double %22, %3, !dbg !224
  %25 = fadd fast double %23, %24, !dbg !224
  store double %25, double* %"'de4", align 8, !dbg !224
  %26 = load double, double* %"'de4", align 8, !dbg !224
  br label %L40, !dbg !224

L24:                                              ; preds = %top
  %27 = icmp sgt i64 %0, -1, !dbg !225
  br i1 %27, label %L31, label %L28, !dbg !226

L28:                                              ; preds = %L24, %top
  %28 = call nonnull {}* @jl_box_int64(i64 signext %0), !dbg !226
  call void @llvm.trap() #6, !dbg !226
  unreachable

L31:                                              ; preds = %L24
  %29 = sitofp i64 %0 to double, !dbg !228
  %30 = call double @llvm.pow.f64(double %3, double %29), !dbg !229
  %31 = fsub fast double %29, 1.000000e+00, !dbg !230
  %32 = call fast double @llvm.pow.f64(double %3, double %31), !dbg !229
  %33 = fmul fast double %29, %32, !dbg !230
  store double %33, double* %"'de5", align 8, !dbg !230
  %34 = icmp slt i64 %0, 21, !dbg !230
  %35 = load double, double* %"'de5", align 8, !dbg !232
  br i1 %34, label %L40, label %L33, !dbg !232

L33:                                              ; preds = %L31
  %36 = call nonnull {} addrspace(10)* bitcast ({}* (i64)* @jl_box_int64 to {} addrspace(10)* (i64)*)(i64 signext %0), !dbg !233
  %37 = call nonnull {} addrspace(10)* bitcast ({}* (i64)* @jl_box_int64 to {} addrspace(10)* (i64)*)(i64 signext %0), !dbg !233
  %38 = call cc38 nonnull {} addrspace(10)* bitcast ({} addrspace(10)* ({} addrspace(10)*, {} addrspace(10)**, i32, {} addrspace(10)*)* @jl_invoke to {} addrspace(10)* ({} addrspace(10)*, {} addrspace(10)*, {} addrspace(10)*, {} addrspace(10)*, {} addrspace(10)*, {} addrspace(10)*)*)({} addrspace(10)* addrspacecast ({}* inttoptr (i64 4586695184 to {}*) to {} addrspace(10)*), {} addrspace(10)* addrspacecast ({}* inttoptr (i64 4634188944 to {}*) to {} addrspace(10)*), {} addrspace(10)* nonnull %36, {} addrspace(10)* addrspacecast ({}* inttoptr (i64 4635819680 to {}*) to {} addrspace(10)*), {} addrspace(10)* nonnull %37, {} addrspace(10)* addrspacecast ({}* inttoptr (i64 4635819776 to {}*) to {} addrspace(10)*)), !dbg !233
  call void @llvm.trap() #6, !dbg !232
  unreachable

L40:                                              ; preds = %L31, %L18, %L14, %top
  %value_phi131720 = phi double [ %30, %L31 ], [ %7, %L14 ], [ %20, %L18 ], [ %3, %top ]
  %"value_phi131720'" = phi fast double [ %35, %L31 ], [ %13, %L14 ], [ %26, %L18 ], [ %6, %top ], !dbg !234
  store double %"value_phi131720'", double* %"value_phi131720'de", align 8, !dbg !234
  %39 = add i64 %0, -1, !dbg !234
  %40 = load i64, i64 addrspace(11)* getelementptr ({ i8 addrspace(13)*, i64, i16, i16, i32 }, { i8 addrspace(13)*, i64, i16, i16, i32 } addrspace(11)* addrspacecast ({ i8 addrspace(13)*, i64, i16, i16, i32 }* inttoptr (i64 4639416368 to { i8 addrspace(13)*, i64, i16, i16, i32 }*) to { i8 addrspace(13)*, i64, i16, i16, i32 } addrspace(11)*), i64 0, i32 1), align 8, !dbg !234, !tbaa !79, !range !84
  %41 = icmp ult i64 %39, %40, !dbg !234
  br i1 %41, label %idxend, label %oob, !dbg !234

L44:                                              ; preds = %idxend, %top
  %value_phi131721 = phi double [ %value_phi131720, %idxend ], [ 1.000000e+00, %top ]
  %"value_phi131721'" = phi fast double [ %68, %idxend ], [ 0.000000e+00, %top ]
  %value_phi3 = phi double [ %phi.cast, %idxend ], [ 1.000000e+00, %top ]
  store double %"value_phi131721'", double* %"value_phi131721'de", align 8, !dbg !216
  %42 = fdiv double %value_phi131721, %value_phi3, !dbg !216
  %43 = load double, double* %"value_phi131721'de", align 8, !dbg !236
  %44 = fmul fast double %43, %value_phi3, !dbg !236
  %45 = fmul fast double %value_phi3, %value_phi3, !dbg !236
  %46 = fdiv fast double %44, %45, !dbg !236
  store double %46, double* %"'de6", align 8, !dbg !236
  %47 = call double @llvm.fabs.f64(double %42), !dbg !236
  %48 = load double, double* %"'de6", align 8, !dbg !238
  %49 = fcmp fast olt double %42, 0.000000e+00, !dbg !238
  %50 = select fast i1 %49, double -1.000000e+00, double 1.000000e+00, !dbg !238
  %51 = fmul fast double %50, %48, !dbg !238
  %52 = fcmp ule double %47, 1.000000e-08, !dbg !238
  br i1 %52, label %L64, label %L52.lr.ph, !dbg !237

L52.lr.ph:                                        ; preds = %L44
  %53 = fmul double %3, %3, !dbg !240
  br label %L52, !dbg !237

L52:                                              ; preds = %L52, %L52.lr.ph
  %iv = phi i64 [ %iv.next, %L52 ], [ 0, %L52.lr.ph ]
  %value_phi526 = phi double [ %42, %L52.lr.ph ], [ %60, %L52 ]
  %value_phi425 = phi double [ %42, %L52.lr.ph ], [ %61, %L52 ]
  %iv.next = add nuw nsw i64 %iv, 1, !dbg !243
  %54 = sitofp i64 %iv.next to double, !dbg !245
  %55 = fdiv double -1.000000e+00, %54, !dbg !249
  %56 = add i64 %iv.next, %0, !dbg !250
  %57 = sitofp i64 %56 to double, !dbg !251
  %58 = fdiv double %55, %57, !dbg !256
  %59 = fmul double %53, %58, !dbg !257
  %60 = fmul double %value_phi526, %59, !dbg !257
  %61 = fadd double %value_phi425, %60, !dbg !258
  %62 = call double @llvm.fabs.f64(double %60), !dbg !236
  %63 = fcmp ule double %62, 1.000000e-08, !dbg !238
  br i1 %63, label %L64.loopexit, label %L52, !dbg !237

L64.loopexit:                                     ; preds = %L52
  %64 = phi i64 [ %iv, %L52 ], !dbg !260
  store i64 %64, i64* %loopLimit_cache, align 8, !dbg !260, !invariant.group !261
  br label %L64, !dbg !260

L64:                                              ; preds = %L64.loopexit, %L44
  %value_phi4.lcssa = phi double [ %42, %L44 ], [ %61, %L64.loopexit ]
  ret double %value_phi4.lcssa, !dbg !260

oob:                                              ; preds = %L40
  call void @llvm.trap() #6, !dbg !234
  unreachable, !dbg !234

idxend:                                           ; preds = %L40
  %65 = load i64 addrspace(13)*, i64 addrspace(13)* addrspace(11)* addrspacecast (i64 addrspace(13)** inttoptr (i64 4639416368 to i64 addrspace(13)**) to i64 addrspace(13)* addrspace(11)*), align 16, !dbg !234, !tbaa !123, !nonnull !4
  %"'il_phi" = phi i64 addrspace(13)* , !dbg !234
  %66 = getelementptr inbounds i64, i64 addrspace(13)* %65, i64 %39, !dbg !234
  %67 = load i64, i64 addrspace(13)* %66, align 8, !dbg !234, !tbaa !125
  %phi.cast = sitofp i64 %67 to double, !dbg !262
  %68 = load double, double* %"value_phi131720'de", align 8, !dbg !262
  br label %L44, !dbg !262

allocsForInversion:                               ; No predecessors!
  %"iv'ac" = alloca i64, align 8
  %loopLimit_cache = alloca i64, align 8
  store i64 0, i64* %loopLimit_cache, align 8
  %"'de" = alloca double, align 8
  store double 0.000000e+00, double* %"'de", align 8
  %"'de1" = alloca double, align 8
  store double 0.000000e+00, double* %"'de1", align 8
  %"'de2" = alloca double, align 8
  store double 0.000000e+00, double* %"'de2", align 8
  %"'de3" = alloca double, align 8
  store double 0.000000e+00, double* %"'de3", align 8
  %"'de4" = alloca double, align 8
  store double 0.000000e+00, double* %"'de4", align 8
  %"'de5" = alloca double, align 8
  store double 0.000000e+00, double* %"'de5", align 8
  %"value_phi131720'de" = alloca double, align 8
  store double 0.000000e+00, double* %"value_phi131720'de", align 8
  %"value_phi131721'de" = alloca double, align 8
  store double 0.000000e+00, double* %"value_phi131721'de", align 8
  %"'de6" = alloca double, align 8
  store double 0.000000e+00, double* %"'de6", align 8
}

  %20 = call double @llvm.fabs.f64(double %19), !dbg !85
Assertion failed: (!isConstantValue(val)), function setDiffe, file /workspace/srcdir/Enzyme/enzyme/Enzyme/GradientUtils.h, line 1792.
```

**Comment by wsmoses** [comment]
Comes from:
```julia
function mybesselj(ν, z, atol=1e-8)
    k = 0
    s = (z/2)^ν / factorial(ν)
    out = s
    while abs(s) > atol
        k += 1
        s *= (-1) / k / (k+ν) * (z/2)^2
        out += s
    end
    out
end
```

**Comment by wsmoses** [comment]
My wager here is that the call/instruction is inactive (or activity analysis incorrectly believes the instruction is active)

**Comment by wsmoses** [comment]
Compiler explorer version for ease https://enzyme.mit.edu/explorer/#g:!((g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:18,endLineNumber:16,positionColumn:18,positionLineNumber:16,selectionStartColumn:18,selectionStartLineNumber:16,startColumn:18,startLineNumber:16),source:'%23include+%3Calgorithm%3E%0A%23include+%3Ccmath%3E%0A%23include+%3Cstring.h%3E%0A%23include+%3Cstdio.h%3E%0A%23include+%3Cassert.h%3E%0A%0Adouble+__enzyme_autodiff(void*,+...)%3B%0A%0Adouble+__enzyme_fwddiff(void*,+...)%3B%0A%0Adouble+f(double+z)%7B%0A++++double+s+%3D+z/2%3B%0A++++double+out+%3D+s%3B%0A++++for(int+k%3D1%3B+std::abs(s)+%3E+1e-10%3B+k%2B%2B)+%7B%0A++++++++s+*%3D+k%3B+%0A++++++++out+%2B%3D+s%3B%0A++++%7D%0A++++return+out%3B%0A%7D%0A%0Aint+enzyme_dupnoneed%3B%0A%0Aint+main()+%7B%0A++++printf(%22%25f%5Cn%22,+__enzyme_fwddiff((void*)f,+enzyme_dupnoneed,+2.0,+1.0))%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:49.71110610861948,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:clang12-enzyme-main,filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'0',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!((name:eigen,ver:'340')),options:'-g0+-S+-emit-llvm+-O0+-ffast-math',selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1,tree:'1'),l:'5',n:'0',o:'clang+12+(cb44221)+(C%2B%2B,+Editor+%231,+Compiler+%231)',t:'0')),k:50.28889389138052,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',m:64.6626752120306,n:'0',o:'',t:'0'),(g:!((h:output,i:(compiler:1,editor:1,fontScale:14,fontUsePx:'0',tree:'1',wrap:'1'),l:'5',n:'0',o:'Output+of+clang+12+(cb44221)+(Compiler+%231)',t:'0')),header:(),l:'4',m:35.33732478796939,n:'0',o:'',s:0,t:'0')),l:'3',n:'0',o:'',t:'0')),version:4

```c
#include <algorithm>
#include <cmath>
#include <string.h>
#include <stdio.h>
#include <assert.h>

double __enzyme_autodiff(void*, ...);

double __enzyme_fwddiff(void*, ...);

double f(double z){
    double s = z/2;
    double out = s;
    for(int k=1; std::abs(s) > 1e-10; k++) {
        s *= k; 
        out += s;
    }
    return out;
}

int enzyme_dupnoneed;

int main() {
    printf("%f\n", __enzyme_fwddiff((void*)f, enzyme_dupnoneed, 2.0, 1.0));
}
```

**Comment by tgymnich** [comment]
This should've been fixed by #399
